src/lib/consensus/proof_of_stake.ml

```diff
@@ -1375,20 +1375,22 @@ let next_proposal now (state : Consensus_state.value) ~local_state ~keypair
         | None -> find_winning_slot (Epoch.Slot.succ slot)
         | Some data -> Some (slot, data)
     in
-    find_winning_slot (Epoch.Slot.succ slot)
+    find_winning_slot slot
```

Comment:
we should track whether or not we have already proposed for this slot or not yet. You can probably stick that information into local state by keeping the last checked epoch and slot.